src/components/Canvas/CanvasTemplate.vue

```diff
@@ -0,0 +1,189 @@
+      currentXPosition: 0,
+      currentYPosition: 0,
+    };
+  },
+  methods: {
+    getNextPositions() {
+      return this.potentialPositions
+        .sort(() => Math.random() - 0.5)
+        .slice(0, 3);
+    },
+    startPosition() {
+      this.isDrag = true;
+    },
+    endPosition() {
+      this.isDrag = false;
+    },
+    touchMove() {
+      if (this.isDrag) {
+        // TODO: Collision analysis
+      }
+    },
+    mouseMove(event) {
+      if (this.isDrag) {
+        // The movement is inversed due to the fact that sign of each movements are inversed between mousewhell and mouse move.
+        this.updatePageSize(
+          -event.movementX,
+          -event.movementY,
+          this.currentXPosition,
+          this.currentYPosition
+        );
+        this.move(event);
+      }
+    },
+    wheelMove(event) {
+      // Calculate the mouse movement and apply it on the current position
+      const xMovement = this.currentXPosition + event.deltaX;
+      const yMovement = this.currentYPosition + event.deltaY;
+      this.updateCurrentPosition(xMovement, yMovement);
+
+      this.updatePageSize(
+        event.deltaX,
+        event.deltaY,
+        event.offsetX,
+        event.offsetY
+      );
+    },
+    move(event) {
+      // Calculate the mouse movement and apply it on the current position
+      const xMovement = this.currentXPosition - event.movementX;
+      const yMovement = this.currentYPosition - event.movementY;
+      this.updateCurrentPosition(xMovement, yMovement);
+
+      // Move
+      window.scrollTo({
+        left: this.currentXPosition,
+        top: this.currentYPosition,
+      });
+    },
+    updateCurrentPosition(newXPosition, newYPosition) {
+      // Ensure that the scroll is in the page
+      if (newXPosition > 0 || newXPosition < this.pageWidth) {
+        this.currentXPosition = newXPosition;
+      }
+      // Ensure that the scroll is in the page
+      if (newYPosition > 0 || newYPosition < this.pageHeight) {
+        this.currentYPosition = newYPosition;
+      }
+    },
+    updatePageSize(deltaX, deltaY, offsetX, offsetY) {
+      // The four next block discovered the movement direction and modify the page size if necessary and also the position of the image if necessary
+      if (deltaY < 0 && offsetY < this.windowHeight) {
+        this.pageHeight = this.pageHeight - deltaY;
+        this.centralImageTopPosition = this.centralImageTopPosition - deltaY;
+      }
+      if (deltaY > 0 && offsetY > this.pageHeight - this.windowHeight) {
+        this.pageHeight = this.pageHeight + deltaY;
+      }
+      if (deltaX < 0 && offsetX > this.windowWidth) {
+        this.pageWidth = this.pageWidth - deltaX;
+        this.centralImageLeftPosition = this.centralImageLeftPosition - deltaX;
+      }
+      if (deltaX > 0 && offsetX > this.pageWidth - this.windowWidth) {
+        this.pageWidth = this.pageWidth + deltaX;
+      }
+    },
+  },
+  computed: {
+    generateNextPositions() {
```

Comment:
since computed variables behave like variables please name them accordingly. For instance `nextPositions()`

Reply:
and you can put the content of the getNextPositions mehod directly here

Reply:
For the name ok but unfortunately, regarding the code, it's not possible because that breaks a side-effect constraints. The code cannot be put directly in the computed variable and must be put outside. It's the reason why a method has been implemented

Reply:
you can do that since you never change this.potentialPositions
```
const potentialPositions =  [1, 2, 3, 4, 5, 6]
      return potentialPositions
        .sort(() => Math.random() - 0.5)
        .slice(0, 3);
    },
```
anyway, the value will never be recomputed since it depends only on initial value so probably does not make so much sense to put it in computed value.
I guess the canvas is only instantiated once so not sure we need random positions here

Reply:
Right it is not necessary to have a computed variable for that. I keep the random in the created function. This random aspect will use later for the next images and the code is already here so easy to move later in the final generation function.